### ISSUE TITLE
Handle non-API errors (auth, net, etc)

### ISSUE THREAD

**Issue by pekeler** [arangodb/arangojs#24]
Right now, if the authentication for the internal request fails, collection.document(id, cb) calls the callback without error and a blank string as data for the document. And cursor.all(cb) throws an exception:
```
foo/node_modules/arangojs/lib/cursor.js:48
      self._index = self._result.length;
                                ^
TypeError: Cannot read property 'length' of undefined
    at foo/node_modules/arangojs/lib/cursor.js:48:33
    at foo/node_modules/arangojs/lib/cursor.js:25:32
    at ArrayCursor.extend._more (foo/node_modules/arangojs/lib/cursor.js:32:25)
    at ArrayCursor.extend._drain (foo/node_modules/arangojs/lib/cursor.js:23:10)
    at ArrayCursor.extend.all (foo/node_modules/arangojs/lib/cursor.js:47:10)
```

**Comment by pekeler** [comment]
I just ran into this again when trying to do some load testing:

```JavaScript
db.collection('As')
  .then(function(collection) {
    return insertGazillionAs(collection);
  }).then(function() {
    return db.collection('Bs')
  }).then(function(collection) {
    return insertGazillionBs(collection);
  }).then(function() {
    return db.collection('Cs')
  }).then(function(collection) {
    return insertGazillionCs(collection);
  }).then(function() {
    console.log('all done :)');
  }).catch(function(error) {
    console.error(error.stack);
  });
```

I forgot to include the http auth params when connection to the DB, and was wondering why I didn't see my documents despite the success message.

**Comment by pluma** [comment]
Fixed in 4.0.0-alpha3.
